src/channel.cpp

```diff
@@ -643,6 +668,13 @@ EGetDataStat CChannel::GetData ( CVector<uint8_t>& vecbyData, const int iNumByte
         {
             // channel is disconnected
             eGetStatus = GS_CHAN_NOT_CONNECTED;
+
+            if ( bDisconnectAndDisable && !bIsServer )
+            {
+                bDisconnectAndDisable = false;
```

Comment:
??! So these get set to the same values whether `eGetStatus` is `GS_CHAN_NOW_DISCONNECTED` or `GS_CHAN_NOT_CONNECTED`?  Now I'm really confused.